### PR TITLE
Updates node dependencies via npm audit

### PR DESCRIPTION
**Pull request by conorsch**
Begins the process of updating the node dependencies based on output from `npm audit`. There's now a CI check in the "safety" workflow that will fail until `npm audit` reports no problems.

See the commit messages for details on what's been done so far. Still a few dangling updates to be made, so requesting collab from @harrislapiroff. Don't hesitate to drop commits in here if you'd prefer to reset and update the deps in a more nuanced fashion.

### CODE REVIEW

**Comment by harrislapiroff** [comment]
As far as I'm concerned this branch is ready! However, since it now contains as much of my code as anyone else's, I'd like to get @chigby's eyes on it—particularly my last few commits converting it from `gulp` to `webpack`. Some testing guidelines:

- [ ] Ensure everything runs without errors locally, using docker-compose
- [ ] Ensure the website javascript and CSS renders
- [ ] Ensure fonts look right and icons look right
- [ ] Ensure you can make changes to javascript and CSS and see those reflected on the site

**Comment by conorsch** [comment]
Took a pass through in the local dev env. Results: 

- [x] Ensure everything runs without errors locally, using docker-compose
- [x] Ensure the website javascript and CSS renders
- [ ] :red_circle: (see below) Ensure fonts look right and icons look right
- [ ] :large_orange_diamond: (did not test) Ensure you can make changes to javascript and CSS and see those reflected on the site

Noticed one regression on the styles, namely that the grades on the leaderboard are no longer colored according to score. See screenshots below.

### On this branch, with webpack
stn-grades-uncolored

### On master branch
stn-grades-colored-2

That was the only problem that was obvious to me. As noted above, I didn't test in-place modifications of the styles—but I'll consider a fix on the style regression here proof that we can indeed change them. :smiley:

**Comment by chigby** [comment]
Looks like the problem @conorsch is seeing is due to changes with string interpolation introduced by pug 2.0, notably that it now uses standard JS string templates:

> We removed support for interpolation in attributes since the implementation was unnecessarily complex, and the feature tended to discourage users from learning that they can just use any JavaScript value in place of attributes. 

See [the migration guide](https://pugjs.org/api/migration-v2.html#attribute-interpolation) for more info.

Basically, we need to make this change:
```diff
--- a/client/src/leaderboardtemplate.jade
+++ b/client/src/leaderboardtemplate.jade
@@ -34,7 +34,7 @@ mixin check(condition)
               a(href=item.absolute_url)= item.name
               if item.pledge
                 a(href=item.pledge.url)
-                  i.pledged.fa.fa-handshake-o(title="#{ item.name } have pledged to secure their site.")
+                  i.pledged.fa.fa-handshake-o(title=`${item.name} have pledged to secure their site.`)
             td.desktop-only
               +check(item.valid_https)
             td.desktop-only
@@ -46,6 +46,6 @@ mixin check(condition)
             td.desktop-only
               +check(item.hsts_preloaded)
             td
-              span(class='grade #{item.grade.class_name}')= item.grade.grade
+              span(class=`grade ${item.grade.class_name}`)= item.grade.grade
```

File: README.md [hunk 1]
```diff
@@ -136,17 +136,6 @@ The test suite includes a smoke test for `createdevdata`, so you can easily
...
-
-The default gulp `watch` task uses `gulp-livereload` to automatically trigger a
-browser refresh when changes to the frontend code are detected. In order to take
-advantage of this, you will need to install the [LiveReload Chrome
-extension](https://chrome.google.com/webstore/detail/livereload/jnihajbhpnppcggbcgedagnkighmdlei?hl=en).
-
-Once you've installed the extension, simply load the development site in your
-browser (`localhost:8000`) and click the LiveReload extension icon to initiate
-live reloading.
-
```
Comment (chigby) [inline README.md]: I didn't personally take advantage of this live reloading feature.. seems a shame, though. I wonder if there are ways to get this same thing in webpack? If not for STN then maybe some of our other sites. It does seem handy and very, you know, "modern web dev toolchain"-y.

File: package.json [hunk 2]
```diff
@@ -47,8 +47,11 @@
   },
   "dependencies": {
     "backbone": "^1.3.3",
+    "file-loader": "^4.0.0",
+    "font-awesome": "^4.7.0",
     "jade": "^1.11.0",
     "jquery": "^3.1.1",
+    "source-sans-pro": "^2.45.0",
```
Comment (chigby) [inline package.json]: This is really cool, that we can just include the font directly from npm.

Comment (harrislapiroff) [reply package.json]: I am honestly like more excited about that than almost anything else in this PR 😁

File: README.md (hunk 1 above)
Comment (harrislapiroff) [reply README.md]: Yeahhhh. There's definitely ways of making this work with Webpack (and with React, there's even ways to _just_ reload the component that gets changed) but I know Emily was fussing with it and never got it to work, so I imagine it's kind of a pain 😬

**Comment by conorsch** [comment]
Thanks, @chigby! One more request. While I can confirm that the change you pushed up resolved the color regression on the full leaderboard display, it still occurs on the homepage (which my initial report did not mention):

### Live site in prod
stn-homepage-leaderboard-color

### On this branch
stn-homepage-leaderboard-nocolor

It's not immediately obvious to me where the corresponding change should be made—can you take a look?

**Comment by conorsch** [comment]
Gorgeous. @chigby's latest changes resolve all styling issues, based on local testing. Looks good!

**Pull request merged**